crates/cli/src/commands/patterns_test.rs

```diff
@@ -273,11 +284,193 @@ pub(crate) async fn run_patterns_test(
 
     let testable_patterns = collect_testable_patterns(patterns);
 
-    if testable_patterns.is_empty() {
-        bail!("No testable patterns found. To test a pattern, make sure it is defined in .grit/grit.yaml or a .md file in your .grit/patterns directory.");
+    get_marzano_pattern_test_results(testable_patterns.clone(), &libs, &arg, flags.clone().into())
+        .await?;
+
+    if arg.watch {
+        let _ = enable_watch_mode(testable_patterns, &libs, &arg, flags.into()).await;
+    }
+    Ok(())
+}
+
+async fn enable_watch_mode(
+    testable_patterns: Vec<GritPatternTestInfo>,
+    libs: &PatternsDirectory,
+    args: &PatternsTestArgs,
+    output: OutputFormat,
+) -> Result<()> {
+    let path = Path::new(".grit");
+    let ignore_path = [".grit/.gritmodules", ".grit/.gitignore", ".log"];
+    // setup debouncer
+    let (tx, rx) = std::sync::mpsc::channel();
+    // notify backend configuration
+    let backend_config = notify::Config::default().with_poll_interval(Duration::from_millis(10));
+    // debouncer configuration
+    let debouncer_config = Config::default()
+        .with_timeout(Duration::from_millis(10))
+        .with_notify_config(backend_config);
+    // select backend via fish operator, here PollWatcher backend
+    let mut debouncer = new_debouncer_opt::<_, notify::PollWatcher>(debouncer_config, tx).unwrap();
+
+    debouncer.watcher().watch(path, RecursiveMode::Recursive)?;
+    log::info!("\n[Watch Mode] Enabled on path: {}", path.display());
+
+    let testable_patterns_map = testable_patterns
+        .iter()
+        .map(|p| (p.local_name.clone().unwrap(), p.clone()))
+        .collect::<HashMap<_, _>>();
+
+    // event processing
+    for result in rx {
+        match result {
+            Ok(event) => {
+                let modified_file_path = event.first().unwrap().path.clone();
+
+                if !modified_file_path.is_file() {
+                    continue;
+                }
+                let modified_file_path = modified_file_path.into_os_string().into_string().unwrap();
```

Comment:
Likely cloning twice unnecessarily.

Reply:
this seems to be the best available way to do it: https://stackoverflow.com/a/42579588

Reply:
Right, but the `clone()` on line 327 shouldn't be needed.

Reply:
yes, it can be changed to:
```rust
                let modified_file_path = &event.first().unwrap().path;

                if !modified_file_path.is_file() {
                    continue;
                }
                let modified_file_path = modified_file_path
                    .clone()
                    .into_os_string()
                    .into_string()
                    .unwrap();
```
which is slightly better because of `if` condition in the middle. but `clone()` can't be avoided as `into_os_string()` requires owned `PathBuf`.